### PR TITLE
Add clang tidy config file and fix a few warnings

### PR DESCRIPTION
**Pull request by seanm**
All the code changes here were done automatically by the tool.

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -31,7 +31,6 @@ modernize-*,\
 -modernize-return-braced-init-list,\
 -modernize-use-auto,\
 -modernize-use-trailing-return-type,\
--modernize-use-emplace,\
```
Comment (dzenanz) [inline .clang-tidy]: Why remove this after you apply it?

**Review by dzenanz — APPROVED** [review]

**Review by aylward — APPROVED** [review]
LGTM!

File: .clang-tidy (hunk 1 above)
Comment (seanm) [reply .clang-tidy]: Because all such warnings are now fixed.

Comment (dzenanz) [reply .clang-tidy]: But they might re-appear in the future? As development continues?

Comment (seanm) [reply .clang-tidy]: The `-` sign indicates it was a suppression of a warning type, and now it's not suppressed.

Comment (dzenanz) [reply .clang-tidy]: Oh, that's what I was missing 🤦

Comment (seanm) [reply .clang-tidy]: Yeah, the syntax is a sorta weird mix in inclusions and exclusions: https://clang.llvm.org/extra/clang-tidy/

**Pull request merged**